### PR TITLE
Serialize functions

### PR DESCRIPTION
**Pull request by rickwierenga**
Alternative to https://github.com/PyLabRobot/pylabrobot/pull/213

Currently used by compute_volume_from_height and compute_height_from_volume. Introduces `allow_marshal` flag on `deserialize` because this is unsafe for untrusted data.

### CODE REVIEW

**Comment by rickwierenga** [comment]
@jkhales what do you think?

File: pylabrobot/resources/petri_dish_tests.py [hunk 1]
```diff
@@ -60,6 +62,7 @@ def test_petri_dish_holder_deserialization_with_dish(self):
     petri_dish_holder = PetriDishHolder("petri_dish_holder")
     petri_dish_holder.assign_child_resource(PetriDish("petri_dish", 90.0, 15.0),
                                             location=Coordinate.zero())
+    print(petri_dish_holder.serialize())
```
Comment (jkhales) [inline pylabrobot/resources/petri_dish_tests.py:65]:
```suggestion
```

File: pylabrobot/resources/resource.py [hunk 2]
```diff
@@ -533,14 +537,15 @@ def deserialize(cls, data: dict) -> Self:
       del data_copy[key]
     children_data = data_copy.pop("children")
     rotation = data_copy.pop("rotation")
-    resource = subclass(**data_copy)
-    resource.rotation = Rotation.deserialize(rotation)
+    resource = subclass(**deserialize(data_copy, allow_marshal=allow_marshal))
+    # resource = subclass(**data_copy)
```
Comment (jkhales) [inline pylabrobot/resources/resource.py:541]:
```suggestion
```

**Review by jkhales — APPROVED** [review]
nice

**Comment by rickwierenga** [comment]
good catches, thanks!

**Pull request merged**